sklearn/compose/_column_transformer.py

```diff
@@ -372,6 +376,53 @@ def _fit_transform(self, X, y, func, fitted=False):
             else:
                 raise
 
+    def _calculate_inverse_indices(self, X):
+        """
+        Private function to calcuate indicies for inverse_transform
+        """
+        # checks for overlap
+        all_indexes = set()
+        input_indices = []
+        for name, trans, cols in self.transformers:
+            col_indices = _get_column_indices(X, cols)
+            col_indices_set = set(col_indices)
+            if not all_indexes.isdisjoint(col_indices_set):
+                self._invert_error = ("Unable to invert: transformers "
+                                      "contain overlaping columns")
+                return
+            if trans == 'drop':
+                self._invert_error = "'{}' drops columns".format(name)
+                return
+            input_indices.append(col_indices)
+            all_indexes.update(col_indices_set)
+
+        # check remainder
+        remainder_indices = self._remainder.indices
+        if (remainder_indices is not None
+                and self._remainder.transformer == 'drop'):
+            self._invert_error = "remainder drops columns"
+            return
+
+        if remainder_indices is not None:
+            input_indices.append(remainder_indices)
+
+        self._input_indices = input_indices
+        self._n_features_in = X.shape[1]
+        self._X_columns = X.columns if hasattr(X, 'columns') else None
+        self._X_is_sparse = sparse.issparse(X)
+        self._invert_error = ""
+        self._output_indices = []
+        cur_index = 0
+
+        Xs = self._fit_transform(X[0:1], None, _transform_one, fitted=True)
```

Comment:
instead of transforming here again, we could also save the dimensions of the outputs in `self.fit_transform` itself?

Reply:
or actually, you can also pass ``Xs`` if we want to keep the code here ?